server-phoenix/lib/helios/widget.ex

```diff
@@ -18,4 +20,18 @@ defmodule Helios.Widget do
 
     belongs_to :location, Helios.Location
   end
+
+  def enabled(query) do
+    from q in query, where: q.enabled == true
+  end
+
+  def available(query, time) when is_binary(time) do
+    td = elem(Timex.parse(time, "{ISO:Extended}"),1)
+    available(query, td)
+  end
+
+  def available(query, time) do
+      tod = DateTime.to_time(time)
+      from q in query, where: (q.start <= ^tod or is_nil(q.start)) and (q.stop >= ^tod or is_nil(q.stop))
```

Comment:
Nit: indentation here. @malssid how are you checking elixir formatting?

Reply:
I haven't been running `mix format` but it's something we should start doing.